### ISSUE TITLE
Rule proposal: `no-unreadable-iife`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1730]
### Description

A realworld case 

```js
const leadingSpaceCount = ((match) => (match ? match[1].length : Number.POSITIVE_INFINITY))(
  content.match(/^( *)[^\n\r ]/m)
);
```

I found it's very unreadable, let's forbid this IIFE with parenthesized arrow function body.

### Fail

```js
const foo = ((bar) => (bar ? bar.baz : baz))(getBar());
```

### Pass

```js
const bar = getBar();
const foo = bar ? bar.baz : baz;
```

**Comment by sindresorhus** [comment]
Accepted
